SwordImportExportPlugin.inc.php

```diff
@@ -0,0 +1,230 @@
+ * @ingroup plugins_generic_sword
+ *
+ * @brief Sword deposit plugin
+ */
+
+import('lib.pkp.classes.plugins.ImportExportPlugin');
+
+class SwordImportExportPlugin extends ImportExportPlugin {
+	/** @var SwordPlugin Parent plugin */
+	protected $_parentPlugin = null; 
+
+	/**
+	 * Constructor
+	 * @param $parentPlugin SwordPlugin
+	 * 
+	 */
+	public function __construct($parentPlugin) {
+		parent::__construct();
+		$this->_parentPlugin = $parentPlugin;
+	}
+
+	/**
+	 * Called as a plugin is registered to the registry
+	 * @param $category String Name of category plugin was registered to
+	 * @return boolean True if plugin initialized successfully; if false,
+	 * 	the plugin will not be registered.
+	 */
+	public function register($category, $path) {
+		$success = parent::register($category, $path);
+		$this->addLocaleData();
+		return $success;
+	}
+
+	/**
+	 * Get reference to the sword plugin
+	 * @return SwordPlugin
+	 */
+	public function getSwordPlugin() {
+		return $this->_parentPlugin;
+	}
+
+	/**
+	 * Get the name of this plugin. The name must be unique within
+	 * its category.
+	 * @return String name of plugin
+	 */
+	public function getName() {
+		return 'SwordImportExportPlugin';
+	}
+
+	/**
+	 * Get the display name.
+	 * @return string
+	 */
+	public function getDisplayName() {
+		return __('plugins.importexport.sword.displayName');
+	}
+
+	/**
+	 * Get the display description.
+	 * @return string
+	 */
+	public function getDescription() {
+		return __('plugins.importexport.sword.description');
+	}
+
+	/**
+	 * @copydoc Plugin::getTemplatePath($inCore)
+	 */
+	public function getTemplatePath($inCore = false) {
+		return $this->getSwordPlugin()->getTemplatePath($inCore);
+	}
+
+	/**
+	 * Display the plugin.
+	 * @param $args array
+	 * @param $request PKPRequest
+	 */
+	public function display($args, $request) {
+		parent::display($args, $request);
+		$templateMgr = TemplateManager::getManager($request);
+		$context = $request->getContext();
+		$journal = $request->getJournal();
+		switch (array_shift($args)) {
+			case 'index':
+			case '':
+				$this->getSwordPlugin()->import('classes.DepositPoint');
+				$depositPointDao = DAORegistry::getDAO('DepositPointDAO');
+				$depositPoints = $depositPointDao->getByContextId($context->getId(), null, SWORD_DEPOSIT_TYPE_MANAGER);
+				$depositPointsData = array('' => __('common.select'));
+				while ($depositPoint = $depositPoints->next()) {
+					$depositPointsData[$depositPoint->getId()] = $depositPoint->getLocalizedName();
+				}
+				$dispatcher = $request->getDispatcher();
+				$settingUrl = $dispatcher->url(
+					$request, ROUTE_PAGE,
+					null, 'management', 'settings', 'website',
+					array(),
+					'swordSettings'
+				);
+				foreach (array('selectedDepositPoint', 'depositEditorial', 'depositGalleys') as $var) {
+					$templateMgr->assign($var, $request->getUserVar($var));
+				}
+				$templateMgr->assign('swordSettingsPageUrl', $settingUrl);
```

Comment:
Just FYI, for lots of variable assignments you can use

    $templateMgr->assign(array(
      'var1' => $var1,
      'var2' => $var2,
    ));

Reply:
refactored.